putter.go

```diff
@@ -260,138 +205,70 @@ func (p *putter) Close() error {
 		p.abort()
 		return p.err
 	}
-	// Complete Multipart upload
-	body, err := xml.Marshal(p.xml)
+
+	eTag, err := p.client.CompleteMultipartUpload(p.uploadID, p.parts)
 	if err != nil {
 		p.abort()
 		return err
 	}
+	p.eTag = eTag
 
-	attemptsLeft := 5
-	for {
-		retryable, err := p.tryPut(body)
-		if err == nil {
-			// Success!
-			break
-		}
+	if err := p.checkMd5sOfParts(); err != nil {
+		return err
+	}
 
-		attemptsLeft--
-		if !retryable || attemptsLeft == 0 {
+	if p.c.Md5Check {
+		md5Path := fmt.Sprint(".md5", p.url.Path, ".md5")
```

Comment:
That extra ".md5" at the beginning looks like a mistake?

Reply:
It's curious, to be sure, but I think it's intended. `p.url.Path` includes a leading `/`, so `md5Path` looks like `".md5/0byte.md5"`. This gets a leading slash added by `Bucket.url()`, resulting in URLs like `"https://<BUCKET>.s3.amazonaws.com/.md5/0byte.md5"`.